src/client.ts

```diff
@@ -239,27 +224,66 @@ in an environmental variable named FAUNA_SECRET or pass it to the Client\
   }
 
   async #query<T = any>(queryRequest: QueryRequest): Promise<QuerySuccess<T>> {
-    const { query, arguments: args } = queryRequest;
-    const headers: { [key: string]: string } = {
-      Authorization: `Bearer ${this.#clientConfiguration.secret}`,
-    };
-    this.#setHeaders(queryRequest, headers);
     try {
-      const result = await this.client.post<QuerySuccess<T>>(
-        "/query/1",
-        { query, arguments: args },
-        { headers }
+      const url = `${this.clientConfiguration.endpoint.toString()}query/1`;
+      const headers = {
+        Authorization: `Bearer ${this.#clientConfiguration.secret}`,
+        // WIP - typecheck should be user configurable, but hard code for now
+        "x-typecheck": "false",
+        // WIP - presently core will default to tagged; hardcode to simple for now
+        // until we get back to work on the JS driver.
+        "x-format": "simple",
+        // include per-request headers
+      };
+      this.#setHeaders(
+        { ...this.clientConfiguration, ...queryRequest },
+        headers
       );
-      const txnDate = new Date(result.data.txn_time);
+
+      const requestData = {
+        query: queryRequest.query,
+        // WIP: encode arguments here
+        arguments: queryRequest.arguments,
+      };
+
+      const fetchResponse = await this.client.request({
+        url,
+        method: "POST",
+        headers,
+        data: requestData,
+      });
+
+      let parsedResponse: HTTPResponse;
+      try {
+        parsedResponse = {
+          ...fetchResponse,
+          // WIP: add decoding here
+          body: JSON.parse(fetchResponse.body),
+        };
+      } catch (error: unknown) {
+        throw new ProtocolError({
+          message: `Error parsing response as JSON: ${error}`,
+          httpStatus: fetchResponse.status,
+        });
```

Comment:
this will always throw a [SyntaxError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#exceptions) if this is not valid JSON.

Let's do a couple things:

1. remove this catch block
2. update `#getError` to map `SyntaxError` to a protocol error.

That will do a couple things:

1. `#getError` remains the only place responsible for producing errors
2. We tighten the rule on what gets lumped as a protocol error for invalid json.

Reply:
One reason this is handled inline with the httpRequest is that the status code is in scope when constructing the ProtocolError. We lose that if we just let the try-catch send the error to `#getError`.

So I am thinking about how we can have a more specific contract with `#getError` so that we know if the argument is pre-request or post-request along with the appropriate context.

Reply:
discussed offline. moving finer distinctions to P2